packages/engine/src/simulation/package/state/packages/behavior_execution/mod.rs

```diff
@@ -249,7 +255,6 @@ impl Package for BehaviorExecution {
         let active_task = self.begin_execution(state_proxy, context, lang).await?;
         let msg = active_task.drive_to_completion().await?;
         // Wait for results
-        // TODO: Get latest metaversions from message and reload state if necessary.
```

Comment:
Do we not need to do this anymore?

Reply:
No, because state can be reloaded at the start of the next package if it needs to read it (in the case of state packages) or before all output/context packages are run. That way, the next package doesn't have to reload state if it doesn't need it.

Reply:
How cheap is `maybe_reload` when it's already up to date? Because it might be safer to just put reloads when we expect they'll be needed. 

I guess it depends how vigilant we are with not using the unchecked methods and such, but I don't think all of our data-access goes through paths that are aware of metaversions so I could still see it being quite easy to make difficult-to-find mistakes due to not reloading first

Reply:
No, as the metaversions are now stored in the memory now directly.

Reply:
The idea is that it might not be up to date, but it might be changed again before we read it, so it would be reloaded twice.

I think everything in Rust outside of the datastore is checking metaversions, but maybe we could do `if cfg!(debug_assertions) { state.write()?.maybe_reload() }`?

Reply:
I don't think we should call `maybe_reload` inside of a conditional compilation because it has side effects, which could lead to surprising behavior.